conn.go

```diff
@@ -3,32 +3,46 @@
 package preconn
 
 import (
+	"bytes"
+	"errors"
+	"io"
 	"net"
 )
 
-// Conn is a net.Conn that supports replaying.
+// Conn is a net.Conn that supports replaying. Reads are not concurrency-safe.
```

Comment:
This was already the case.  If we want to make this concurrency safe, I can do so.  I just figured I'd hold off on adding any extra complexity since it was already not concurrency safe.  Parallel reading from connections is an unlikely use-case anyway.